tests/pytorch/test_dataloader.py

```diff
@@ -71,14 +71,18 @@ def test_saint(num_workers, mode):
 @parametrize_idtype
 @pytest.mark.parametrize('mode', ['cpu', 'uva_cuda_indices', 'uva_cpu_indices', 'pure_gpu'])
 @pytest.mark.parametrize('use_ddp', [False, True])
-def test_neighbor_nonuniform(idtype, mode, use_ddp):
+@pytest.mark.parametrize('use_mask', [False, True])
+def test_neighbor_nonuniform(idtype, mode, use_ddp, use_mask):
     if mode != 'cpu' and F.ctx() == F.cpu():
         pytest.skip('UVA and GPU sampling require a GPU.')
+    if mode != 'cpu' and use_mask:
+        pytest.skip('Masked sampling only works on CPU.')
```

Comment:
Then do we need to perform device check and raise error in the related APIs?

Reply:
We have a check in `CSRRowWiseSampling` in `src/array/array.cc`.